### PR TITLE
Added Sample Cron Job to check working of Whenever Gem

### PR DESCRIPTION
**Pull request by namangupta01**
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!
Follow-up pr of #2706 
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] PR body includes `fixes #0000`-style reference to original issue #
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by jywarren** [comment]
Great! @namangupta01 you should be able to run `git push -f https://github.com/publiclab/plots2.git unstable` and this will build at http://jenkins.laboratoriopublico.org/, and then you can check it out at https://unstable.publiclab.org/cron_log.log

However, you'd probably want to move the log file into a world-readable directory -- `/public/cron_log.log` -- that way you'll be able to confirm it works using a browser!

**Comment by namangupta01** [comment]
Done!

**Comment by Gauravano** [comment]
hey @namangupta01 , are you able to open the link provided by @jywarren?

**Comment by namangupta01** [comment]
I moved it to the public folder.
Have a look here!
https://github.com/publiclab/plots2/commit/6583d609720bb136be77e03423780f57010440f2

**Comment by namangupta01** [comment]
We need to run command `whenever --update cronjob`. Where should i write this command?

**Comment by jywarren** [comment]
I'm not sure, maybe @icarito knows?

File: Dockerfile [hunk 1]
```diff
@@ -28,7 +28,12 @@ ADD Gemfile /tmp/Gemfile
 ADD Gemfile.lock /tmp/Gemfile.lock
 RUN bundle install --jobs 4
 
+
+
 ADD . /app
 WORKDIR /app
 
+RUN bundle exec whenever --update-crontab
+CMD service cron start
```
Comment (icarito) [inline Dockerfile]: Dockerfile is only run at image build time. It's not right to start the cron service here. It should be run at deployment time. We are in the process of automating that. I'll add a `.Jenkinsfile` today and we can add these lines there. Currently the deployment script is edited directly at our Jenkins instance. Please revert edits to this file.

**Comment by namangupta01** [comment]
@icarito I have reverted the edits. Have you edit the `Jenkinsfile` file? So i can deploy it to the unstable and can test it out.

**Comment by icarito** [comment]
Hey @namangupta01 sorry I took so long I was looking at a complex solution https://jenkins.io/doc/book/pipeline/jenkinsfile/ while a simple one was already available (Makefile). Pushed to unstable and we can see what it needs...

**Comment by namangupta01** [comment]
@icarito  Hi, the build at unstable is failed! Showing the following :
>[EnvInject] - Variables injected successfully.
[plots2] $ /bin/sh -xe /tmp/jenkins4012669757992778900.sh
> +make build/tmp/jenkins4012669757992778900.sh: 2: /tmp/jenkins4012669757992778900.sh: make: not found
Build step 'Execute shell' marked build as failure
Finished: FAILURE

**Comment by icarito** [comment]
Yes I'm debugging it - I need to install make in Jenkins container rather than plots container.

**Comment by icarito** [comment]
Hi,
Here I've included the necessary tweaks to Makefile and Dockefile in order to correctly build the image and deploy in unstable. Sorry for the delay, it conflated with the upgraded reference from the `ruby-2.3` docker image tag.

**Comment by namangupta01** [comment]
Oh Great! So now we can test it into unstable.

**Comment by jywarren** [comment]
Anything you'd like to fix within the CodeClimate stuff? I know it's being quite strict, but it ought to result in cleaner code even if it's not totally related to your work. Your choice!

**Comment by jywarren** [comment]
And if you could just resolve the final conflict in the Makefile, we should be good!

**Comment by jywarren** [comment]
There's currently an issue with Coveralls -- http://status.coveralls.io/ -- but this looks good.

**Comment by jywarren** [comment]
🎉 Merging!!!

**Comment by jywarren** [comment]
Hm, https://publiclab.org/cron_log.log seems empty? How often ought this to run?  Or do we need to run:

`whenever --update cronjob` ?

**Comment by icarito** [comment]
Yes. Previously we didn't have this step in deployment and then we implemented deploying with Makefile.
Basically we need to add https://github.com/publiclab/plots2/blob/master/Makefile#L27 and https://github.com/publiclab/plots2/blob/master/Makefile#L28 after line 16.
It should be safe to run these commands manually now in production to see it work.

There are a [couple of Make tasks](https://github.com/publiclab/plots2/blob/master/Makefile#L30) that @ryzokuken added last year, are these used or is it safe to remove them?

**Comment by jywarren** [comment]
@namangupta01 can you open a pull request with @icarito's suggested changes to the Makefile?

**Comment by namangupta01** [comment]
sure! Sorry for delaying this.

**Pull request merged**